src/pks/mpc/mpc_coupled_dualmedia_water.hh

```diff
@@ -0,0 +1,87 @@
+/*
+  ATS is released under the three-clause BSD License. 
+  The terms of use and "as is" disclaimer for this license are 
+  provided in the top-level COPYRIGHT file.
+
+  Authors: Daniil Svyatsky (dasvyat@lanl.gov)
+*/
+//! A coupler which integrates surface, richards and preferantial flows implicitly.
+
+/*!
+
+
+
+*/
+
+#ifndef PKS_MPC_COUPLED_DUALMEDIA_WATER_HH_
+#define PKS_MPC_COUPLED_DUALMEDIA_WATER_HH_
+
+#include "Operator.hh"
+#include "mpc_delegate_water.hh"
+#include "pk_physical_bdf_default.hh"
+#include "TreeOperator.hh"
+#include "strong_mpc.hh"
+
+namespace Amanzi {
+
+class MPCCoupledDualMediaWater : public StrongMPC<PK_BDF_Default> {
+ public:
+
+
+  MPCCoupledDualMediaWater(Teuchos::ParameterList& FElist,
+                  const Teuchos::RCP<Teuchos::ParameterList>& plist,
+                  const Teuchos::RCP<State>& S,
+                  const Teuchos::RCP<TreeVector>& soln);
+
+  virtual void Setup(const Teuchos::Ptr<State>& S);
+  virtual void Initialize(const Teuchos::Ptr<State>& S);
+
+  virtual void set_states(const Teuchos::RCP<State>& S,
+                          const Teuchos::RCP<State>& S_inter,
+                          const Teuchos::RCP<State>& S_next);
+
+  // -- computes the non-linear functional g = g(t,u,udot)
+  //    By default this just calls each sub pk FunctionalResidual().
+  virtual void FunctionalResidual(double t_old, double t_new, Teuchos::RCP<TreeVector> u_old,
+           Teuchos::RCP<TreeVector> u_new, Teuchos::RCP<TreeVector> g);
+
+  // -- Apply preconditioner to u and returns the result in Pu.
+  virtual int ApplyPreconditioner(Teuchos::RCP<const TreeVector> u, Teuchos::RCP<TreeVector> Pu);
+
+  // -- Update the preconditioner.
+  virtual void UpdatePreconditioner(double t, Teuchos::RCP<const TreeVector> up, double h);
+
+ protected:
+  // void
+  // UpdateConsistentFaceCorrectionWater_(const Teuchos::RCP<const TreeVector>& u,
+  //         const Teuchos::RCP<TreeVector>& Pu);
+
+ protected:
+
+  Teuchos::RCP<Operators::TreeOperator> op_tree_matrix_, op_tree_pc_;
```

Comment:
Is this ever used?  I see you create and initialize this, but I don't see you ever use either of these?